frontend/Dockerfile

```diff
@@ -4,13 +4,12 @@ FROM node:lts AS builder
 # set working directory
 WORKDIR /app
 
-COPY package.json yarn.lock /app/
-RUN yarn install
-
 # add app
 COPY . /app/
 
-RUN yarn --verbose build
+RUN yarn install
```

Comment:
Moving `yarn install` here will increase image build times.

The reason `package.json` and `yarn.lock` are copied first is to cache the dependency installation step when rebuilding the image.

Any particular reason you moved `yarn install` here?

Reply:
Yes, I got some errors:

![image](https://user-images.githubusercontent.com/5595092/109056061-41957d80-76e0-11eb-90e4-28eca2dfa0ef.png)

If you add this command before the build step: `RUN yarn esbuild --help`, it produces this error (`Exec format error`)

![image](https://user-images.githubusercontent.com/5595092/109056192-6ab60e00-76e0-11eb-865a-538af3d92621.png)

I had the exact same issue during the `yarn bsb -make-world` step but this time with the `ppx`. I've pretty much never used docker before, but my understanding is that `COPY . /app/` will copy over the whole project, including the `node_modules`. Within this `node_modules` are some executables like `ppx` and `esbuild`. I guess that these executables are prebuilt for each platform, and during the `postinstall` step, the one corresponding to the host system (macOS in my case) is installed.

Since the `docker` image is a `linux` one, these executables can't run. Hence why I moved the `yarn install` step after we copy the project.

That being said, I don't know how docker cache works, so there might be a better solution ofc.

Reply:
Note that I ran this command: `docker build --no-cache -t frontend .` which explicitly disables the cache. But I guess it should work even without cache?

Reply:
Quick tip before a more complete answer: try removing your local `node_modules` before building the image.

We should add a `.dockerignore` file.

Reply:
Oh yes, removing `node_modules` makes sense and looks to work! :) I moved back the `yarn install` step and added a `.dockerignore` file in 14bed96ce77001efc3f994a05fb67bd85db88307